rpc/rest/wiring_impl.go

```diff
@@ -0,0 +1,161 @@
+				return err
+			}
+			if err := p.Wire(WithLog(overwrite));err != nil {
+				return err
+			}
+			if err := p.Wire(WithPluginConfig(overwrite));err != nil  {
+				return err
+			}
+			return nil
+		}
+		return errors.Errorf("Could not convert core.Plugin to *%s.Plugin",packageName)
+	}
+	return ret
+}
+
+// WithLog returns a wiring that sets the Log dependency of the Plugin
+// If overwrite is false, WithLog will leave unchanged any existing Log wired in
+// If a log value is provided, it will be wired in, otherwise, a default will be
+// generated and wired in
+func WithLog(overwrite bool,log...logging.PluginLogger) wiring.Wiring {
+	ret := func (plugin core.Plugin) error {
+		p, ok := plugin.(*Plugin)
+		if ok {
+			if overwrite || p.Log == nil {
+				if len(log) > 0 {
+					p.Log = log[0]
+				} else {
+					p.Log = logging.ForPlugin(p.Name(), logrus.NewLogRegistry())
+				}
+			}
+			return nil;
+		}
+		return errors.Errorf("Could not convert core.Plugin to *%s.Plugin",packageName)
+	}
+	return ret
+}
+
+// WithLogFactory wires a Log dependency generated from the LogFactory (if one is provided)
+// If overwrite is false, existing values will not be overwritten
+// If a factory is provided, it will be used, otherwise, a default LogFactory will be generated and used
+func WithLogFactory(overwrite bool,factory...logging.LogFactory) wiring.Wiring {
+	ret := func (plugin core.Plugin) error {
+		p, ok := plugin.(*Plugin)
+		if ok {
+			err := p.Wire(WithLog(overwrite,logging.ForPlugin(p.Name(), logrus.NewLogRegistry())))
+			return err
+		}
+		return errors.Errorf("Could not convert core.Plugin to *%s.Plugin",packageName)
+	}
+	return ret
+}
+
+// WithName wires a PluginName for the Plugin
+// If overwrite is false, existing values will not be overwritten
+// If name is provided, that will be configured as the PluginName, otherwise a default will be used
+func WithName(overwrite bool,name...string) wiring.Wiring {
+	ret := func (plugin core.Plugin) error {
+		p, ok := plugin.(*Plugin)
+		if ok {
+			if overwrite || p.PluginName == "" {
+				if len(name) > 0 {
+					p.PluginName = core.PluginName(name[0])
+				} else {
+					p.PluginName = core.PluginName(defaultName)
+				}
+			}
+			return nil
+		}
+		return errors.Errorf("Could not convert core.Plugin to *%s.Plugin",packageName)
+	}
+	return ret
+}
+
+// WithPluginConfig wires in a PluginConfig for the plugin
+// If overwrite is false, existing values will not be overwritten
+// If cfg is provided, that will be configured as the PluginConfig, otherwise a default will be used
+func WithPluginConfig(overwrite bool, cfg... config.PluginConfig) wiring.Wiring{
+	ret := func (plugin core.Plugin) error {
+		p, ok := plugin.(*Plugin)
+		if ok {
+			if len(cfg) > 0 {
+				p.PluginConfig = cfg[0]
+			} else {
+				if err := p.Wire(WithName(false)); err != nil {
+					return err
+				}
+				if overwrite || p.PluginConfig == nil {
+					p.PluginConfig = config.ForPlugin(p.Name())
```

Comment:
Missing extra parameter here makes example fail with error that the flag was not defined. Although it shouldn't be needed because it seems it's just a bug. 
See here: https://github.com/ligato/cn-infra/blob/master/config/plugin_config.go#L70
The if condition there fails because `len(opts)` is zero and that prevents registering the flag.